odbc/src/api/functions.rs

```diff
@@ -769,57 +769,51 @@ pub extern "C" fn SQLGetDiagRecW(
     // Make the record number zero-indexed
     let rec_number = (rec_number - 1) as usize;
     match handle_type {
-        HandleType::Env => match unsafe { (*mongo_handle).as_env() } {
-            Some(env) => {
-                let env_contents = (*env).read().unwrap();
-                match env_contents.errors.get(rec_number) {
-                    Some(odbc_err) => util::get_diag_rec(
-                        odbc_err,
-                        state,
-                        message_text,
-                        buffer_length,
-                        text_length_ptr,
-                        native_error_ptr,
-                    ),
-                    None => SqlReturn::NO_DATA,
-                }
+        HandleType::Env => {
+            let env = check_invalid! {unsafe { (*mongo_handle).as_env() }};
+            let env_contents = (*env).read().unwrap();
+            match env_contents.errors.get(rec_number) {
```

Comment:
I think we should actually put add a HandlesContents trait or some such name with an `errors` method so we can drastically simplify this function too.

Reply:
Unfortunately, because of the way LockGuards work there is no way to make this much better. We can use a macro, but the macro invocation is almost as big as the code already, so it only adds complexity.